internal/cmd/onepasswordtemplatefuncs.go

```diff
@@ -58,19 +61,26 @@ func (c *Config) onepasswordItemFieldsTemplateFunc(args ...string) map[string]in
 }
 
 func (c *Config) onepasswordDocumentTemplateFunc(args ...string) string {
+	sessionToken := c.onepasswordGetOrRefreshSession(args)
 	onepasswordArgs := getOnepasswordArgs([]string{"get", "document"}, args)
-	output := c.onepasswordOutput(onepasswordArgs)
+	output := c.onepasswordOutput(onepasswordArgs, sessionToken)
 	return string(output)
 }
 
-func (c *Config) onepasswordOutput(args []string) []byte {
+func (c *Config) onepasswordOutput(args []string, sessionToken string) []byte {
 	key := strings.Join(args, "\x00")
 	if output, ok := c.Onepassword.outputCache[key]; ok {
 		return output
 	}
 
+	var secretArgs []string
+	if sessionToken != "" {
+		secretArgs = []string{"--session", sessionToken}
+	}
+
 	name := c.Onepassword.Command
-	cmd := exec.Command(name, args...)
+	// appending the session token here, so we don't log it by accident
+	cmd := exec.Command(name, append(secretArgs, args...)...)
```

Comment:
Instead of passing the session token on the command line (which also makes it visible to other users on the same machine via `ps`) is it possible to pass the session token in an environment variable instead?

Reply:
Not that I know of, unfortunately. I hope [1Password implements that feature](https://1password.community/discussion/126053/reuse-a-session-token-from-environment-not-from-command-line/).

What we could do in the meantime, is add a config option to chezmoi that will toggle this whole thing.
Two open questions I have about that are:
- Where and how would such a setting be configured?
- Should it default to enabled or disabled?

Reply:
Good point. I agree that putting this behavior behind a switch makes sense: the key breaking change it introduces is that it changes chezmoi from exiting with an error when no 1Password token is available to prompting the user for interactive input, which will break scripts written with no interaction in mind. However, for almost all users, I think it is a clear improvement in behavior so it should be enabled by default.

I would add a `onepassword.prompt` boolean [configuration variable](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#variables) defaulting to `true`.

Please could you also update the [documentation for the `onepassword` template function](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#onepassword-uuid-vault-uuid-account-name) to describe the behavior.

Thank you!

Reply:
By the way, I'm also happy to accept this PR as-is and implement the above myself if you prefer.